### PR TITLE
added AttributeError in subs

### LINKED ISSUE #8886: Symbol subs Vector exception
```
A = ReferenceFrame('A')
v = A.x
s, t = symbols('s t')
```

```(s / t)```
output: s / t

```(v / t)```
output: 1/t n̂x 

```(s / t).subs(s, v)```
output:
```
anaconda3/lib/python3.4/site-packages/sympy/parsing/sympy_parser.py in eval_expr(code, local_dict, global_dict)
    689     """
    690     expr = eval(
--> 691         code, global_dict, local_dict)  # take local objects in preference
    692 
    693     return expr

<string> in <module>()

AttributeError: 'function' object has no attribute 'x'
```

### LINKED ISSUE #8892: subs needs a better error message when trying to sub in Vectors
See issue  #8886. User tried to substitute a `sympy.physics.vector.Vector` for a scalar symbol in an expression. The error message isn't that informative. It should tell you that it isn't allowed.

### PR DESCRIPTION
**Pull request by mvnnn**
Fixes #8886

closes #8899 
closes #8892

### CODE REVIEW

File: sympy/core/basic.py [hunk 1]
```diff
@@ -842,6 +843,10 @@ def subs(self, *args, **kwargs):
                    When a single argument is passed to subs
                    it should be a dictionary of old: new pairs or an iterable
                    of (old, new) tuples."""))
+        elif isinstance(args[1], Vector) and self.find(args[0]):
```
Comment (smichr) [inline sympy/core/basic.py]: I forget now how this is handled, but I think this should be handled by creating a pertinent `_eval_subs` routine in the relevant object. I think we had to do something like this with Integral (by not letting the dummy variable be targeted by substitution). Putting it there keeps the Basic routine from becoming cluttered with special cases.

Comment (mvnnn) [inline sympy/core/basic.py]: for integral i will add this type of condition. that's right.
```
>>> Integral(y, (y, 1, 3)).subs(y, x)
AttributeError: dummy variable substitution is not allowed
>>> Integral(y+x, (y, y+1, 3)).subs(y, x)
Integral(y+x, (y, x+1, 3)).subs(y, x)
```

Comment (mvnnn) [inline sympy/core/basic.py]: @smichr please check the changes.

**Comment by mvnnn** [comment]
@smichr please check the changes.if that's ok then i will doing same thing for integral.

**Comment by mvnnn** [comment]
@smichr please check the changes.

File: sympy/core/basic.py [hunk 2]
```diff
@@ -843,7 +843,9 @@ def subs(self, *args, **kwargs):
                    it should be a dictionary of old: new pairs or an iterable
                    of (old, new) tuples."""))
         elif len(args) == 2:
-            sequence = [args]
+            _args_valid = Basic._eval_subs(self, args[0], args[1])
```
Comment (smichr) [inline sympy/core/basic.py]: no change here

**Comment by smichr** [comment]
Two notes:

1. In your OP, the auto-closing syntax requires you to say 'closes, fixes, or resolves' (or similar without trailing 's') followed by the issue number. By including 'for' between the two, the syntax is not correct and the issue will not be closed when this is committed.

1. I put a debugging session in the issue #8886 which you should see. Perhaps it will generate more discussion. Your fix here, I see, is superficial and doesn't get to the core reason that the subs is failing. Let's see what the better solution is after others read what I wrote, ok? Maybe you will have an idea of what to do next, too.

**Comment by smichr** [comment]
:eyeglasses: See [here for auto-closing syntax](https://help.github.com/articles/closing-issues-via-commit-messages/)

**Comment by mvnnn** [comment]
@smichr thanks, i will try to implement this.

**Comment by smichr** [comment]
Please see what you think about my 8886 branch which I will open as a friendly replacement of this one.

**Comment by smichr** [comment]
Does #10574 look ok as an alternative to you?

**Comment by mvnnn** [comment]
@smichr one vector subs by other vector is also possible.i think you need to add this.

**Pull request closed without merging**